civihr_employee_portal/civihr_employee_portal.module

```diff
@@ -1680,6 +1671,43 @@ function civihr_employee_portal_menu() {
     'type' => MENU_NORMAL_ITEM,
   ];
 
+  // Base structure for a menu item which renders a webform asynchronously
+  $webform_popup_base_menu_structure = [
+    'page callback' => 'civihr_employee_portal_webform_render_callback',
+    'access callback' => TRUE,
+    'type' => MENU_CALLBACK,
+  ];
+
+  // Create Emergency Contact webform ajax popup
+  $items['create-emergency-contact/js/view'] = [
+    'page arguments' => ['Create Emergency Contact'],
+  ] + $webform_popup_base_menu_structure;
+
+  // Create Dependant webform ajax popup
+  $items['create-dependant/js/view'] = [
+    'page arguments' => ['Create Dependant'],
+  ] + $webform_popup_base_menu_structure;
+
+  // My Details - My Home Address webform ajax popup
+  $items['edit-my-home-address/js/view'] = [
+    'page arguments' => ['My Home Address'],
+  ] + $webform_popup_base_menu_structure;
+
+  // My Details - My Payroll webform ajax popup
+  $items['edit-my-payroll/js/view'] = [
+    'page arguments' => ['My Payroll'],
+  ] + $webform_popup_base_menu_structure;
+
+  // My Details - Personal Details webform ajax popup
+  $items['edit-my-personal-details/js/view'] = [
+    'page arguments' => ['My Personal Details'],
+  ] + $webform_popup_base_menu_structure;
+
+  // My Details - Contact Info webform ajax popup
+  $items['edit-my-contact-details/js/view'] = [
```

Comment:
i think the creation of all these webforms modal should be encapsulated in a private function, something like

```php
function civihr_employee_portal_menu() {
  // ...

  _civihr_employee_portal_webform_modal($items, 'create-emergency-contact', 'Create Emergency Contact');
  _civihr_employee_portal_webform_modal($items, 'create-dependant', 'Create Dependant');
  _civihr_employee_portal_webform_modal($items, 'edit-my-personal-details', 'My Personal Details');

  // ...

  return $items;
}

/**
 * Lorem ipsum
 *
 * @param {foo} $bar
 * ...
 */
function _civihr_employee_portal_define_webform_modal(&$items, $path, $name) {
  $items["{$path}/js/view"] = [
    'page arguments' => [$name],
    'page callback' => 'civihr_employee_portal_webform_render_callback',
    'access callback' => TRUE,
    'type' => MENU_CALLBACK,
  ];
}
```

Reply:
Sure. Thanks.